engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1750,23 +1774,28 @@ private IR.Name buildNameOrQualifiedName(Tree t) {
   }
   private List<IR.Name> buildNames(Tree t, char separator, boolean fail) {
     List<IR.Name> segments = nil();
-    for (;;) {
+    LOOP: for (;;) {
       switch (t) {
         case Tree.OprApp app -> {
           if (!String.valueOf(separator).equals(app.getOpr().getRight().codeRepr())) {
             throw new UnhandledEntity(t, "buildNames with " + separator);
           }
-          segments = cons(buildName(app.getRhs()), segments);
-          t = app.getLhs();
+          if (app.getRhs() instanceof Tree.Ident) {
+            segments = cons(buildName(app.getRhs()), segments);
+            t = app.getLhs();
+          } else {
+            segments = cons(buildName(app.getLhs()), segments);
+            t = app.getRhs();
+          }
         }
         case Tree.Ident id -> {
           segments = cons(buildName(id), segments);
-          return segments;
+          break LOOP;
```

Comment:
I don't think it is justified to use labels here, right? I mean you **can** use it but somewhere a unicorn will die.

Reply:
There is one benefit associated with the `LOOP` - now there is just a *single* return statement in the method. I thought you'd love that!

Reply:
The use of label+`break` is not appropriate in current state (there was a middle step that would deserve it, but it got removed at the end). Returned to two `return` statements in [3a10f43](https://github.com/enso-org/enso/pull/3825/commits/3a10f432f44ea0a15640893a80c17a68d0a8a61a).